nova/virt/vmwareapi/vmops.py

```diff
@@ -2772,7 +2913,133 @@ def live_migration(self, instance, migrate_data, volume_mapping):
             config_spec.device = device
             device_config_spec.append(config_spec)
 
-        vm_util.relocate_vm(self._session, vm_ref, spec=relocate_spec)
+        try:
+            vm_util.relocate_vm(self._session, vm_ref, spec=relocate_spec)
+        except vexc.VimException as e:
+            target.delete_config_drive_files(context, instance, new_cdroms)
+            raise e
+
+        self.delete_config_drive_files(context, instance, original_cdroms)
+
+        if new_cdroms:
+            # Technically a post-live-migration task, but we have no
+            # way of passing those changes without API changes to the
+            # destination compute-host
+            try:
+                target.reconfigure_vm_device_change(context, instance,
+                                                    new_cdroms)
+            except messaging.RemoteError:
+                # Failing now would stop the instance reflecting
+                # the new host
+                LOG.exception("Failed to reconfiguring cdroms. "
+                              "Swallowing error as we can't go back now",
+                               instance=instance)
+
+    def delete_config_drive_files(self, ctxt, instance, cdroms):
+        for cdrom_spec in cdroms:
+            try:
+                datastore = getattr(cdrom_spec.device.backing, 'datastore')
+                if not datastore:
+                    continue
+                file_name = getattr(cdrom_spec.device.backing, 'fileName')
+                if not file_name:
```

Comment:
These `getattr()` will raise `AttributeError`. They need an additional `None` like the others.

Reply:
Fixed